.github/workflows/build-test.yml

```diff
@@ -116,6 +116,48 @@ jobs:
         command: make
         args: ci-job-test
 
+  # ci-job-test-cargo
+  test-cargo:
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, macos-latest, windows-latest ]
```

Comment:
I don't think multi-OS adds much value. I don't want to add more macOS CI because the number of macOS workers is more limited than ubuntu.

Reply:
I was thinking about that but I think that this is one where we probably really do want to make sure multi-OS works, because this is testing how real developers will interact with the project, and real developers are more likely to be on these other OSes.

Reply:
Multi-OS works because
* We already run `cargo test` on all OSes
* We already test datagen on all OSes
* We can assume `cargo` works on all OSes

I don't think we have to test this, and in fact the buffer-example apparently is not even OS-agnostic.

We only have 5 concurrent macOS workers. Currently each PR uses two (test and testdata), if we add a third even two PRs will slow each other down.

Reply:
I think we don't thoroughly test datagen `--from-bin` on all OS on CI

Also please note that the latest version of this PR only runs the full test on a daily cronjob

Is there anything actionable here?